### ISSUE TITLE
Default Firebase App Not Initialized, but not clear how to add package explicitly [Android]

### ISSUE THREAD

**Issue by AdamBebko** [TobiasBuchholz/Plugin.Firebase#78]
> If you receive an error that states the default Firebase App is not initialized, adding one package explicitly seems to resolve this issue (it doesn't seem to matter which package gets added).

I am getting this error but am not to sure what it means to "add the package explicitly". 

I have the Plugin.Firebase package installed via nuget, and everything works great in iOS. In Firebase Console I can see the analytics register for IOS when I open the app. But android is giving this error at runtime, and no Firebase Analytics seem to register at all.

![Screen Shot 2022-10-04 at 1 09 59 AM](https://user-images.githubusercontent.com/43276915/193739238-f9b9df71-a20c-4797-9117-663ec8a80678.png)

here are some possibly related lines a few lines beforehand.

![Screen Shot 2022-10-04 at 12 57 14 AM](https://user-images.githubusercontent.com/43276915/193737740-508cbded-988b-412d-9eb4-7900f9ba58c6.png)

I tried adding Xamarin.Firebase.Analytics package 120.1.2 via Nuget. Didn't seem to do anything. Here is relevant entry in CSPROJ

        <ItemGroup Condition="'$(TargetFramework)' == 'net6.0-android'">
		<GoogleServicesJson Include="google-services.json" />
		<PackageReference Include="Xamarin.Firebase.Analytics" Version="120.1.2" />
	</ItemGroup>

Any help explaining what "adding the package explicitly" means would be helpful! 

Details: .Net Maui, running in android emulator on Mac

On iOS, I get the following message, no errors : `8.10.0 - [Firebase/Analytics][I-ACS023007] Analytics v.8.10.0 started`

**Comment by TobiasBuchholz** [comment]
> If you receive an error that states the default Firebase App is not initialized, adding one package explicitly seems to resolve this issue (it doesn't seem to matter which package gets added).

This is only needed if you are using the Plugin.Firebase.Legacy package, but as far as I understand it you are using the newest version 1.2.0, so this step isn't necessary. 

The message `the default Firebase App is not initialized` could also mean that something is wrong with you google-services.json setup. Did you put it in the root of your project directory?

**Comment by AdamBebko** [comment]
Thanks for the quick reply. Yes it's in my root directory.

I have this in my CSPROJ file 

```xml
        <!--This includes the google services plist-->
	<ItemGroup Condition="'$(TargetFramework)' == 'net6.0-ios'">
		<BundleResource Include="GoogleService-Info.plist" />
	</ItemGroup>

	<!--This includes the google services json-->
	<ItemGroup Condition="'$(TargetFramework)' == 'net6.0-android'">
		<GoogleServicesJson Include="google-services.json" />
	</ItemGroup>
```

It seems MAUI is finding the plist on iOS with no problems, but not the json on android. I'm running android in the emulator. Could that be the issue? 

I can't think of anything else.

**Comment by AdamBebko** [comment]
I Finally figured it out. Exasperated, I created a fresh app, and tried to find the source of the issue. The issue was that the package name (i.e. com.me.myapp) was different in the CSPROJ file compared to the google services json's package's. I guess Visual Studio puts in some filler there, rather than the correct package name.

I was able to view debug events from analytics and everything now. Thanks so much for the awesome package. Looking forward to playing around with Firestore now.

Sorry for the noob mistake. If I get some time, I will do a PR to add that info in the setup readme to help future MAUI noobs like me.
